include/alpaka/kernel/TaskKernelOacc.hpp

```diff
@@ -0,0 +1,313 @@
+                getWorkDiv<Block, Threads>(*this));
+            auto const threadElemExtent(
+                getWorkDiv<Thread, Elems>(*this));
+
+#if ALPAKA_DEBUG >= ALPAKA_DEBUG_MINIMAL
+            std::cout << "m_gridBlockExtent=" << this->m_gridBlockExtent << "\tgridBlockExtent=" << gridBlockExtent << std::endl;
+            std::cout << "m_blockThreadExtent=" << this->m_blockThreadExtent << "\tblockThreadExtent=" << blockThreadExtent << std::endl;
+            std::cout << "m_threadElemExtent=" << this->m_threadElemExtent << "\tthreadElemExtent=" << threadElemExtent << std::endl;
+#endif
+
+            // Get the size of the block shared dynamic memory.
+            auto const blockSharedMemDynSizeBytes(
+                meta::apply(
+                    [&](ALPAKA_DECAY_T(TArgs) const & ... args)
+                    {
+                        return
+                            getBlockSharedMemDynSizeBytes<
+                                AccOacc<TDim, TIdx>>(
+                                    m_kernelFnObj,
+                                    blockThreadExtent,
+                                    threadElemExtent,
+                                    args...);
+                    },
+                    m_args));
+
+#if ALPAKA_DEBUG > ALPAKA_DEBUG_MINIMAL
+            std::cout << __func__
+                << " blockSharedMemDynSizeBytes: " << blockSharedMemDynSizeBytes << " B" << std::endl;
+#endif
+            // The number of blocks in the grid.
+            TIdx const gridBlockCount(gridBlockExtent.prod());
+            // The number of threads in a block.
+            TIdx const blockThreadCount(blockThreadExtent.prod());
+
+            if(gridBlockCount == 0 || blockThreadCount == 0)
+            { //! empty grid is a NOP
+                return;
+            }
+
+#if ALPAKA_DEBUG >= ALPAKA_DEBUG_MINIMAL
+            std::cout << "threadElemCount=" << threadElemExtent[0u]
+                << "\tgridBlockCount=" << gridBlockCount << std::endl;
+#endif
+            auto argsD = m_args;
+            auto kernelFnObj = m_kernelFnObj;
+            dev.makeCurrent();
+            #pragma acc parallel num_workers(blockThreadCount) copyin(threadElemExtent,blockThreadExtent,argsD,gridBlockExtent) default(present)
+            {
+                {
+                    #pragma acc loop gang
+                    for(TIdx b = 0u; b<gridBlockCount; ++b)
+                    {
+                        CtxBlockOacc<TDim, TIdx> blockShared(
+                            gridBlockExtent,
+                            blockThreadExtent,
+                            threadElemExtent,
+                            b,
+                            blockSharedMemDynSizeBytes);
+
+                        // Execute the threads in parallel.
+
+                        // Parallel execution of the threads in a block is required because when syncBlockThreads is called all of them have to be done with their work up to this line.
```

Comment:
That line seems to long.

Reply:
clang-format will care about this ;-)

Reply:
fixed, although the exact length was no decided on yet.